File: Java/lambda/2/main.java
/**
 * Created by fay on 2017/5/31.
 */

public class main {
    public static void main(String args[]) {
        main main = new main();
        main.say((content) -> {
            System.out.println(content);
            return "Hello Java!";
        });
    }

    void say(Reply reply) {
        System.out.println(reply.r("Hello World!"));
    }
}

interface Reply {
    String r(String content);
}

File: Lua/lambda/2/main.lua
--
-- Created by IntelliJ IDEA.
-- User: fay
-- Date: 2017/6/15
-- Time: 下午2:25
-- To change this template use File | Settings | File Templates.
--

function main()
    say(reply)
end

finish = function(content)
    print(content)
end

reply = function(finish)
    print("Hello Lua!")
    finish("^_^")
end

function say(reply)
    print("Hello World!")
    reply(finish)
end

main()
